### ISSUE TITLE
Release requires Go 1.21 and no direct proxy

### ISSUE THREAD

**Issue by jpkrohling** [open-telemetry/opentelemetry-collector-contrib#29290]
### Describe the issue you're reporting

When doing the release for v0.89.0, I ran into two problems that could have been avoided by using the proper versions and configuration of Go:

- There's an issue with `github.com/openshift/api` having references to non-existing tags. This is fixed by not using a direct connection to the repositories. A proxy should be used instead by running this command: `go env -w GOPROXY='https://proxy.golang.org,direct'`
- Using Go 1.20.x has a problem resolving a specific dependency (`cloud.google.com/go/compute/metadata`). Using Go 1.21 bypasses this (see below).
- I have also had a go checksum mismatch error related to a module within `go-openapi`. Removing my cache for modules on that namespace solved the problem (`sudo rm -rf /home/jpkroehling/go/pkg/mod/github.com/go-openapi/`), even though I'm certainly not pleasing security-minded folks (why have there been checksum changes??). 

The `github.com/openshift/api` problem is the following:
```
 go: downloading github.com/openshift/api v3.9.0+incompatible
github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/k8snode imports
	github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig imports
	github.com/openshift/client-go/quota/clientset/versioned imports
	github.com/openshift/client-go/quota/clientset/versioned/typed/quota/v1 imports
	github.com/openshift/api/quota/v1: reading github.com/openshift/api/go.mod at revision v3.9.0: unknown revision v3.9.0
```

The `cloud.google.com/go/compute/metadata` problem is the following:
```
github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor tested by
	github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor.test imports
	go.opentelemetry.io/collector/receiver/otlpreceiver imports
	go.opentelemetry.io/collector/config/configgrpc imports
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc tested by
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.test imports
	google.golang.org/grpc/interop imports
	golang.org/x/oauth2/google imports
	cloud.google.com/go/compute/metadata: ambiguous import: found package cloud.google.com/go/compute/metadata in multiple modules:
	cloud.google.com/go v0.26.0 (/home/jpkroehling/go/pkg/mod/cloud.google.com/go@v0.26.0/compute/metadata)
	cloud.google.com/go/compute/metadata v0.2.4-0.20230617002413-005d2dfb6b68 (/home/jpkroehling/go/pkg/mod/cloud.google.com/go/compute/metadata@v0.2.4-0.20230617002413-005d2dfb6b68)
make[2]: *** [../../Makefile.Common:186: tidy] Error 1
make[2]: Leaving directory '/home/jpkroehling/Projects/github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor'
```

**Comment by bryan-aguilar** [comment]
Regarding the goproxy setting, I did a quick search but couldn't find any issues or PRs on the reasoning that we chose direct. I do remember the discussion in SIG meetings though. I am one of the users that must use `GOPROXY=direct` due to company restrictions around the usage of the official go proxy. 

Dealing with checksum mismatches has been a real pain. Usually we have to address those with replace statements due to the underlying tag being changed. A big no no from what I understand in the go packaging world. 

> cloud.google.com/go/compute/metadata

I thought we had replace statements already to handle this error? Maybe it was a past version and we were able to get rid of the replace, because I don't see it in the ADOT collector anymore. https://github.com/aws-observability/aws-otel-collector/issues/926 was this the same issue you were seeing?
